### PR TITLE
feat: developer container environment

### PR DESCRIPTION
**Pull request by vsoch**
Problem: there is no easy way to develop (in a container) and this would be nice if we consider other kinds of bindings. Solution: add a .devcontainer setup.

This is the first step (I think at least) to thinking about Go bindings. I see this is a project in C++ (not C) so I'll look into good ways to do that.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      "settings": {
+        "terminal.integrated.defaultProfile.linux": "bash"
+  },
+      "extensions": [
+         "ms-vscode.cmake-tools"
+      ]
+    }
+  },
+  "postStartCommand": "git config --global --add safe.directory /workspaces/flux-accounting"
+}
```
Comment (cmoussa1) [inline .devcontainer/devcontainer.json:17]: does it matter if this line does or doesn't have a newline at the end of it?

**Review by cmoussa1 — APPROVED** [review]
this LGTM! thanks @vsoch! just had one minor question about `devcontainer.json`

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (vsoch) [reply .devcontainer/devcontainer.json:17]: It does not - it looks like json but actually it breaks many rules of it. I tested this fully and the newline does not seem to make a difference.

Comment (vsoch) [reply .devcontainer/devcontainer.json:17]: hey you have a new picture! I like it!

Comment (cmoussa1) [reply .devcontainer/devcontainer.json:17]: thanks! my hair is a lot longer now, i don't look much like that graduation photo from 2019 anymore 😆 feel free to set MWP on this whenever you are ready

Comment (vsoch) [reply .devcontainer/devcontainer.json:17]: Done! Thanks @cmoussa1

**Pull request merged**